client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js

```diff
@@ -11,122 +11,274 @@ import {
+        .mapValues(row => _.chain(row)
+          .tail()
+          .map(value => parseInt(value))
+          .value()
+        )
+        .value();
+    
+      return population_values_by_prov_code;
+    }
+    ); 
 };
 
-const prepare_data_for_a11y_table = (data) =>
-  _.chain(provinces)
-    .map((province, prov_code) => {
-      if( !_.includes(["onlessncr", "qclessncr", "ncr"], prov_code)){
-        const formatted_yearly_tp = _.map(
-          data, 
-          (row) => formats["compact2_written_raw"](row[prov_code])
-        );
-
-        return {
-          label: province.text,
-          data: formatted_yearly_tp,
-        };
-      }
-    })
-    .filter((data) => !_.isUndefined(data))
-    .value();
 
-export const declare_tp_by_region_panel = () => declare_panel({
-  panel_key: "tp_by_region",
-  levels: ["gov", "dept"],
-  panel_config_func: (level, panel_key) => ({
-    depends_on: ['orgTransferPaymentsRegion'],
-    calculate: function(subject, info, options){
-      const { orgTransferPaymentsRegion } = this.tables;
-      return calculate_common(subject, orgTransferPaymentsRegion);
-    },
-    
-    render(render_args){
-      const {
-        calculations,
-        footnotes,
-        sources,
-      } = render_args;
-      const { panel_args } = calculations;
-      const { data } = panel_args;
-      const current_year_data = _.last(data);
-
-      const largest_prov = _.chain(current_year_data)
+const prepare_data_for_a11y_table = (data) => _.chain(data)
+  .flatMap( _.keys )
+  .uniq()
+  .map( (prov_code) => {
+    const formatted_data = _.map(
+      data,
+      (row) => formats["compact2_written_raw"](row[prov_code] || 0)
+    );
+  
+    return {
+      label: provinces[prov_code].text,
+      data: formatted_data,
+    };
+  } )
+  .filter('data')
+  .value();
+
+
+class TPMap extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: true,
+      population: {},
+    };
+  }
+  componentDidMount(){
+    loadPopulation()
+      .then( 
+        (transformed) => this.setState({
+          loading: false,
+          population: transformed,
+        })
+      );
+  }
+  render(){
+    const { calculations, footnotes, sources } = this.props;
+    const { loading, population } = this.state;
+
+    if (loading) {
+      return (
+        <div style = {{position: "relative", height: "80px", marginBottom: "-10px"}}>
+          <SpinnerWrapper config_name={"sub_route"} />
+        </div>);
+    } else {
+      const { tables } = calculations.panel_args;
+      const { tp } = tables;
+
+      const get_subject_data_for_year = (year) => tp.prov_code(
+        year,
+        calculations.subject.level === 'dept' && calculations.subject.id
+      );
+
+      const data_tp = std_years.map(get_subject_data_for_year);
+      const data_tppc = std_years.map((year, i) => {
+        const single_year_tp_data = get_subject_data_for_year(year);
+        const result = _.chain(
```

Comment:
i think this would look better:
```
        return _.chain( _.keys(single_year_tp_data) )
          .pullAll(["na", "abroad"])
          .map((prov) => {
            const in_year_prov_transfer_payments = single_year_tp_data[prov];
            const in_year_prov_population = population[prov][i];
            return [
              prov,
              in_year_prov_transfer_payments/in_year_prov_population,
            ];
          })
```

Reply:
tested and it still works
fixed 👍